packages/core/src/storages/request_queue_v2.ts

```diff
@@ -235,6 +248,8 @@ export class RequestQueue extends RequestProvider {
                 try {
                     await this.client.deleteRequestLock(id);
                 } catch {
+                    // TODO Kuba: There is raise condition here, if you lock the request and did not unlock it, it will be locked there and run stuck.
+                    // TODO Kuba: We should at least log this.
```

Comment:
This only happens under very specific circumstances, and even if we don't manage to unlock the request, the lock will time out in ~60 seconds, right? So we shouldn't be risking a deadlock.

Reply:
60 seconds in case there is no pageTimeout set in hours which happens in google maps scraper.
I think at least warning or debug log could be there to have some idea to debug this.

Reply:
Well, maybe it's a bad idea to lock requests for this long then. The `RequestQueue` should be able to lock for a shorter time and prolong periodically.

Reply:
Yeah, this was the thing I was proposing in some Slack tread to do it, but I think the output of the discussion is that it can generate even more issues and can be hard to implement as you need to do prolong solid (should be done in the separate tread to avoid block by scraping processes etc).

Reply:
That brings us back to the "please make sure I have N locked requests, prolong them and send them to me" endpoint. If that's the only thing we need to do to have a properly managed queue head, it's worth the extra overhead of making sure that it's not blocked by random stuff on the event loop.

But even if somebody runs something that can hog the event loop for 60s straight (I know, under some circumstances, a shorter time would also be problematic), I'm pretty confident that they will have bigger problems than losing locks on their requests.

Reply:
From meeting:
not strong opinion, but the prolonging lock using interval looks better in that case.